src/arrayWithLength.js

```diff
@@ -47,6 +47,27 @@ const arrayWithLengthFactory = ({
     return null
 }
 
+/**
+ * Ensure the prop value is an array with a length between a minimum and maximum.
+ * If a third `propType` argument is passed each item in the array needs to be of that prop-type
+ * @param {number} [min=0] - The minimal array length
+ * @param {number} [max=Infinity] - The maximal array length
+ * @param {function} [propType] - The prop-type that each array item needs to conform to
+ * @return {Error|null} Returns null if all conditions are met, or an error
+ * @example
+ * import React from 'react'
+ * import { arrayWithLength } from '@dhis2/prop-types'
+ *
+ * const LotsOfLists = props => <div {...props}>Does nothing</div>
+ *
+ * LotsOfLists.propTypes = {
+ *     arrayWithMaxThreeNumbers: arrayWithLength(0, 3, propTypes.number),
+ *     arrayWithAtLeastSixStrings: arrayWithLength(6, undefined, propTypes.string),
+ *     arrayWithAtLeastTenItems: arrayWithLength(10),
+ *     mandatoryArrayBetweenOneAndTen: arrayWithLength(1,10).isRequired,
+ * }
+ * @function
```

Comment:
I'd prefer to see this above the first `@param`, did that not work?
Seems weird to find it after `@example`

Reply:
Me too, but for some reason the documentation got mangled if I move it up there...
Without this annotation these functions are classified as global constants, which is true, but also a bit misleading. So another solution could be to actually rewrite these functions to actual `function` syntax....

Reply:
Changing to `function () {}` allows us to drop the `@function` prop.

Reply:
ed75c0c